sklearn/_loss/loss.py

```diff
@@ -725,6 +731,19 @@ class HalfTweedieLoss(BaseLoss):
     """
 
     def __init__(self, sample_weight=None, power=1.5):
+        check_scalar(
+            power,
+            "power",
+            target_type=numbers.Real,
+            include_boundaries="neither",
+            min_val=-np.inf,
+            max_val=np.inf,
+        )
+        if not isinstance(power, numbers.Number) or not math.isfinite(power):
```

Comment:
With the `check_scalar`, we can remove this custom check.

Reply:
I was working too quick. Will fix it soon.